File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.c
int sumOddLengthSubarrays(int* arr, int arrSize){
    int ans = 0;
    
    for(int i = 0; i < arrSize; i++){
        int sum = 0 , len = 0;
        for(int j = i; j < arrSize; j++){
            len++;
            sum += arr[j];
            if(len % 2 == 1) //odd 
                ans += sum;
         }
    }
    return ans;
}

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.swift
class Solution {
    func sumOddLengthSubarrays(_ arr: [Int]) -> Int {
        var res = 0
        var n = arr.count;
        var i = 0
        
        while i < n {
            res += ((i + 1) * (n - i) + 1) / 2 * arr[i]
            i += 1
        }
        return res
    }
}